### ISSUE TITLE
Zero123 results generates inconsistent results compared to official repo

### ISSUE THREAD

**Issue by y22ma** [threestudio-project/threestudio#246]
Hi!

Thanks for all the great work! I have had good results running the following command written in the README:
```python launch.py --config configs/zero123.yaml --train --gpu 0 data.image_path=./load/images/dog1_rgba.png```

The problem shows up when I introduce my own images. Here's an input image I have and also the result.
![input](https://github.com/threestudio-project/threestudio/assets/3324929/881d9a98-62e9-4bb0-9308-bb3f77cd3091)

https://github.com/threestudio-project/threestudio/assets/3324929/8edf77b8-83fd-43dd-9c52-3d16681887a5

The couch geometry is turned 180 degree, and this implementation of zerio123 forcefully slaps the input image as a texture onto the back of the couch.

Meanwhile, here's the view synthesized via the official zero123 demo on huggingface:
![Screenshot from 2023-07-21 23-03-04](https://github.com/threestudio-project/threestudio/assets/3324929/96ba2445-ae77-4336-999d-4b0934a83cce)

The huggingface demo produces sensible view synthesis, without a 180 degree turn on the geometry. Any idea why there's a difference in performance? Would an initialization via monocular depth image and enabling of depth loss constrain the problem better?

I would be more than happy to take on debugging myself. Would really appreciate commands to run zero123-XL for view synthesis only to see where the pipeline breaks down.

**Comment by voletiv** [comment]
@y22ma In my experiments, I found that the cause is Zero123XL. I believe the model is trained to give such a signal as to invent a new sofa at the back. With a different Zero123 model trained on limited data, or finetuned with LoRA, maybe you'll find better results. This is definitely not a bug in the code :P

**Comment by y22ma** [comment]
Thanks @voletiv for the feedback. Not sure if this is the right place to ask this question, but is there a place where I can read about how threestudio's zero123 pipeline actually work?

It seems that zero123 takes an image as input and outputs a NERF which is then used for view synthesis or 3D mesh extraction via marching cubes (or newer techniques like baked sdf).

So in this case, you're saying that the NERF generated by the zero123 XL from a forward pose is turned 180deg? Would I be able to test view synthesis of zero123 XL with threestudio as well to dig into why there's such a rotation?

**Comment by y22ma** [comment]
Thanks for the feedback @voletiv. Would love to dive deeper to see why zero123XL is behaving this way. It seems that the loss function used in this case is based on SDS from dreamfusion, and not in pixel space, which explains why it's different from viewsynth, which involves the zero123 decoder.

To investigate it further, I will need to turn on guidance_eval in the configuration and record the zero123XL output from its decoder?

**Comment by voletiv** [comment]
@y22ma I believe the HF demo is based on Zero123, and not Zero123XL.

You may get a better result in this particular case using the Zero123 checkpoint instead of Zero123XL. Try it out!
